os/kernel/mqueue/mq_send.c

```diff
@@ -149,6 +150,12 @@ int mq_send(mqd_t mqdes, FAR const char *msg, size_t msglen, int prio)
 	/* mq_send() is a cancellation point */
 	(void)enter_cancellation_point();
 
+	if (mq_check(mqdes) != OK) {
+		leave_cancellation_point();
+		set_errno(EINVAL);
+		return ERROR;
+	}
+
```

Comment:
This check needs to be done in all mq apis. Please check timedsend and timed receive.